### ISSUE TITLE
[Bug]: aws_sfn_state_machine - Produced an invalid new value for .publish

### ISSUE THREAD

**Issue by smokentar** [hashicorp/terraform-provider-aws#32826]
### Terraform Core Version

1.5.4

### AWS Provider Version

5.10.0

### Affected Resource(s)

aws_sfn_state_machine

### Expected Behavior

Step function is updated correctly

### Actual Behavior

Terraform sometimes succeeds and sometimes fails.
This issue occurs intermittently.

I can only reproduce by applying terraform multiple times. (reproduced on every second or third apply)

### Relevant Error/Panic Output Snippet

```shell
Error: Provider produced inconsistent final plan
 
 When expanding the plan for
 module.<placeholder>-us-east-1.module.step-function.aws_sfn_state_machine.this
 to include new values learned so far during apply, provider
 "registry.terraform.io/hashicorp/aws" produced an invalid new value for
 .publish: was cty.False, but now null.
 
 This is a bug in the provider, which should be reported in the provider's
 own issue tracker.
```

### Terraform Configuration Files

```
resource "aws_sfn_state_machine" "this" {
  name     = var.name
  role_arn = aws_iam_role.this.arn
  publish  = false
  definition = templatefile("${path.module}/step-function-definition.tpl",
    {
      name               = var.name
      aws_region         = var.aws_region
      service_account_id = var.service_account_id
      lambda_function_a  = var.lambda_function_names[0]
      lambda_function_b  = var.lambda_function_names[1]
      lambda_function_c  = var.lambda_function_names[2]
      lambda_function_d  = var.lambda_function_names[3]

      step_function_a = var.step_function_names[0]
  })
}
```

Documentation mentions `publish` is optional and defaults to `false`.
I've tried to hardcode it in the config file however issue still occurs.

### Steps to Reproduce

Apply terraform multiple times

### Debug Output

Plan output:
```
# module.<placeholder>-us-east-1.module.step-function.aws_sfn_state_machine.this will be updated in-place
  ~ resource "aws_sfn_state_machine" "this" {
      ~ definition    = jsonencode(
            <cut_json_for_readability>
        ) -> (known after apply)
        id            = "arn:aws:states:us-east-1:<placeholder>:stateMachine:<placeholder>"
        name          = "<placeholder>"
      + publish       = false
        tags          = {}
        # (7 unchanged attributes hidden)

        # (2 unchanged blocks hidden)
    }
```

### References

The `publish` argument was released in 5.6.0 - [PR](https://github.com/hashicorp/terraform-provider-aws/pull/32176)

### Would you like to implement a fix?

None

**Comment by smokentar** [comment]
I can confirm the issue gets resolved when downgrading the AWS provider to 5.5.0

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/pull/32176.
